xds/internal/xdsclient/transport/grpctransport/grpctransport_ext_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package grpctransport_test
+
+import (
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/internal/grpctest"
+	internalbootstrap "google.golang.org/grpc/internal/xds/bootstrap"
+	"google.golang.org/grpc/xds/internal/xdsclient/internal"
+	"google.golang.org/grpc/xds/internal/xdsclient/transport"
+	"google.golang.org/grpc/xds/internal/xdsclient/transport/grpctransport"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+// Tests that the grpctransport.Builder creates a new grpc.ClientConn every time
```

Comment:
Yeah I just meant that in the case where you want all clusters, it states it was returning an empty slice but it was returning nil instead (which happened to also be what was being returned in error cases).

Reply:
No, why would it panic if it is not successfully connected? In fact, I don't even have a real management server running as part of this test. What I meant was if I passed `nil` for the first return value in the overridden client hook, that would cause the code to panic because it calls `cc.Connect` on it.

> So it's testing Build/NewClient come 1:1?

Sort of. But it does not test that `NewClient` actually ends up establishing a connection, because that would mean that we are testing `grpc.NewClient` instead of this `Build` function.